_posts/2024-12-18-Management-Console-on-OpenShift.adoc

```diff
@@ -0,0 +1,77 @@
+
+* Management user: +
+  The management console is protected by default and requires a management user added with the `add-user.sh` script.
+* Public route to the management interface: +
+  The management interface has to be publicly available from the outside.
+* Allowed origin: +
+  The console uses the https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API[fetch API] to talk to the management interface of a running WildFly instance. In an OpenShift environment, the origins of the public route and the management interface itself are different. That's why we need to tell WildFly that it is ok to make requests to the management interface from another origin (see https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS[CORS policies] for more details).
+
+You can build such an image on your own based on the official WildFly images available at https://quay.io/repository/wildfly/wildfly[quay.io/wildfly/wildfly] (see "Extending the image"). Another way is to use the pre-built images from https://quay.io/repository/halconsole/wildfly[quay.io/halconsole/wildfly]. These images are mainly meant for HAL development and testing, but already meet these requirements, which makes them suitable for our use case. In particular the images add a management user `admin:admin` and have a list of preconfigured allowed origins.
+
+To add the allowed origin for the public route, we make use of the https://jmesnil.net/weblog/2024/12/11/a-kubectl-plugin-for-wildfly-jboss-cli/[`jboss-cli`] `kubectl` plugin. This plugin makes it straightforward to connect to a WildFly instance running on OpenShift and execute CLI commands. Please visit https://jmesnil.net/weblog/2024/12/11/a-kubectl-plugin-for-wildfly-jboss-cli/ to find out how to install and use the plugin.
+
+== Instructions
+
+The steps below assume you have access to an OpenShift cluster and installed `kubectl` and the `jboss-cli` plugin.
+
+. Create application
++
+[source,shell]
+----
+oc new-app quay.io/halconsole/wildfly
+oc create route edge --service wildfly --port 9990
+----
+
+. Add allowed origin
++
+Use `oc get pods` to find the name of the pod to connect to and `oc get routes` to get the hostname of the public route to the management interface.
++
+[source,shell]
+----
+kubectl jboss-cli -p <pod>
+----
++
+Login using `admin:admin` and execute these CLI commands:
++
+[source,shell]
+----
+/core-service=management/management-interface=http-interface:list-add(name=allowed-origins,value=https://<hostname>)
+reload
+exit
+----
+
+. Open the management console at `+https://<hostname>+` and login using `admin:admin`.
+
+'''
+
+NOTE: As an alternative to adding the allowed origin, you can also use of the latest online version of the management console available at https://hal.github.io/console.
```

Comment:
I'm not sure if the format is correct: Should the NOTE also contains the background section?

I was kind of expecting to see something like:

```
[NOTE]
====

....
Give an arbitrary name, select *https* as scheme, enter the hostname of the public route _without_ https and port *80*:
 +
 image::hal/add-management-interface.png[Add management interface]
 . Click *Add* and then *Connect*
 . Login using `admin:admin
====
```

Alternatively, you could add a dedicated section about https://hal.github.io/console that contains all the note, background & instructions?

Reply:
I agree. I reformatted that section.